src/post/schemas/post.schema.ts

```diff
@@ -0,0 +1,46 @@
+import { HydratedDocument } from 'mongoose';
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+
+export type PostDocument = HydratedDocument<Post>;
+
+@Schema()
+export class Post {
+  @Prop({ required: true, unique: true })
+  postId: string;
+
+  @Prop({ required: true })
+  title: string;
+
+  @Prop({ required: true })
+  imageUrl: string;
+
+  @Prop({ required: true })
+  description: string;
+
+  @Prop({ required: true })
+  tags: string[];
+
+  @Prop({ required: true})
+  points: number;
+
+  @Prop({ required: true })
+  userId: string;
+
+  @Prop({required: true })
+  numOfLike: number;
+
+  @Prop ({required: true })
+  postedDate: Date;
+
+  @Prop ()
+  dueDate: Date;
+
+  @Prop ({required: true })
+  completed: boolean;
+
+  @Prop()
+  likedUserList: String[];
```

Comment:
string[]
I am also quite confused about String and string, found this on stack overflow
https://stackoverflow.com/questions/14727044/what-is-the-difference-between-types-string-and-string